csharp/src/Google.Protobuf/Collections/CollectionExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace Google.Protobuf.Collections
+{
+
+    /// <summary>
+    /// Extension Methods for Collections
+    /// </summary>
+    public static class CollectionExtensions
```

Comment:
Extension may be misunderstand as we are going to add proto2 support which has extensions. We will also add ExtensionCollection similar with FieldCollection.

Reply:
And please add test for this PR